paddle/platform/CMakeLists.txt

```diff
@@ -1,14 +1,8 @@
 add_subdirectory(dynload)
 
-nv_test(cuda_test SRCS cuda_test.cu DEPS dyload_cuda)
+nv_test(cuda_test SRCS cuda_test.cu)
 
 cc_library(place SRCS place.cc)
 cc_test(place_test SRCS place_test.cc DEPS place glog gflags)
-IF(WITH_GPU)
-    set(GPU_CTX_DEPS dyload_cuda dynamic_loader )
-ELSE()
-    set(GPU_CTX_DEPS)
-ENDIF()
 
-cc_library(device_context SRCS device_context.cc DEPS place eigen3 ${GPU_CTX_DEPS})
```

Comment:
Make device_context a library is a better idea, otherwise, each library which deps on `device_context` should handle complex GPU libraries.